### PR TITLE
Create MVC subfolders

### PR DESCRIPTION
**Pull request by ericthewry**
Hey @sf-interns/owners.  This creates a few directories we might need going into the future. I though it would be handy to get this up there ASAP as we are all going to be writing files, and it would be good to have an organizational structure in place from the get-go.

Note: the `.keep` files are 1 bit files used so that git will recognize the containing folder and allow me to commit it.

Let me know if you have any thoughts.

Here are the newly created folders
- `[app|tests]/services`
- `[app|tests]/controllers`
- `[app|tests]/views`
- `[app|tests]/models`
- `[app|tests]/view_controllers`

I've also modified the paths throughout, and provided an app path variable for the test suite. Let me know if you have any comments or questions.

### CODE REVIEW

File: tests/mycha-test-helper.coffee [hunk 1]
```diff
@@ -6,4 +6,6 @@ global.chai = chai
 global.expect = chai.expect
 global.sinon = sinon
 
+global.app_path = '../../app'
+
 process.env.NODE_ENV = 'test'
```
Comment (bmwwoo) [inline tests/mycha-test-helper.coffee]: I thought sublime automatically added a newline...

**Comment by bmwwoo** [comment]
:shaved_ice: LGTM

**Comment by shubhamgogna** [comment]
Can we add descriptions for `services`, `controllers`, `views`, `models`, and `view_controllers` for later reference? Other than that, LGTM.

**Comment by ericthewry** [comment]
@s-gogna Something like this?

"This app uses an extended model of the [traditional MVC](https://upload.wikimedia.org/wikipedia/commons/a/a0/MVC-Process.svg) design pattern, adding a Service Layer, and View Controllers:

Services allow for an additional level of abstraction between the Controller and the Models, and the View Controllers (sort of like React) provide a way to manipulate the views, taking the information from the models and processing it as needed.

Services, which are sort of part of the M portion of MVC, perform tasks that would traditionally be done by the Controller.  They comprise of the business logic of the application. This allows the Controllers to concern themselves with sending and receiving messages and information and not with logic.  

Models, then, should mostly concern themselves with allowing other parts of the program to manipulate their state (and if there is a database, with updating that state to the database)

Views often the visual representation of a model.

View Controllers is sort of like a service for the views.  It takes the information from the controller and then manipulates the views.  It also provides the visual user interface, passing the information to the controller, which translates it into `back-end` commands.  Depending on what front-end library we use this may be unnecessary."

**Pull request merged**